yt_dlp/extractor/tencent.py

```diff
@@ -63,7 +70,9 @@ def _get_video_api_response(self, video_url, video_id, series_id, subtitle_forma
 
     def _extract_video_formats_and_subtitles(self, api_response, video_id):
```

Comment:
How about something like this for this function?
```py
    def _extract_video_formats_and_subtitles(self, api_response, video_id):
        video_response = api_response['vl']['vi'][0]

        formats, subtitles = [], {}
        for video_format in video_response['ul']['ui']:
            if video_format.get('hls') or determine_ext(video_format['url']) == 'm3u8':
                fmts, subs = self._extract_m3u8_formats_and_subtitles(
                    video_format['url'] + traverse_obj(video_format, ('hls', 'pt'), default=''),
                    video_id, 'mp4', fatal=False)

                formats.extend(fmts)
                self._merge_subtitles(subs, target=subtitles)
            else:
                formats.append({
                    'url': f'{video_format["url"]}{video_response["fn"]}?vkey={video_response["fvkey"]}',
                    'ext': 'mp4',
                })

        identifier = video_response.get('br')
        format_response = traverse_obj(
            api_response, ('fl', 'fi', lambda _, v: v['br'] == identifier),
            expected_type=dict, get_all=False) or {}
        has_drm = video_response.get('drm') == 1
        is_hdr = format_response.get('name') == 'hdr10'
        format_name = f'{format_response.get("sname")} ({format_response.get("resolution")})'
        common_info = {
            'width': video_response.get('vw'),
            'height': video_response.get('vh'),
            'abr': float_or_none(format_response.get('audiobandwidth'), scale=1000),
            'vbr': float_or_none(format_response.get('bandwidth'), scale=1000),
            'fps': format_response.get('vfps'),
            'format': format_name,
            'format_id ': format_response.get('formatdefn'),
            'format_note': f'{format_name} HDR' if is_hdr else None,
            'dynamic_range': 'hdr10' if is_hdr else None,
            'has_drm': has_drm,
        }
        for f in formats:
            f.update(common_info)

        return formats, subtitles
```
It seems to pass the tests. I believe this to ba a lot more readable though.

As a side note, why was the property called `has_drm ` with a space? I have removed the space. The tests would need to be updated accordingly.

Reply:
I'm not a python programmer, so I'm not familiar with many functions of python. Your code looks fine though.

Reply:
I have used these codes to optimize readability and fix whitespace errors in `has_drm`.